### PR TITLE
[backend/frontend] massive delete operations for public dashboard and workspaces (#6509)

### PR DESCRIPTION
**Pull request by lndrtrbn**
### Proposed changes

* Massive delete for public dashboards
* Massive delete for custom dashboards
* Massive delete for investigations

### Related issues

* #6509 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

**Comment by lndrtrbn** [comment]
We have a known issue on duplicating/deleting workspaces and currently looking for a fix
EDIT : has been fixed ✅

File: opencti-platform/opencti-front/src/private/components/workspaces/Workspaces.tsx [hunk 1]
```diff
@@ -46,23 +57,106 @@ const Workspaces: FunctionComponent<WorkspacesProps> = ({
...
+        ]}
+        createButton={isFeatureEnable('FAB_REPLACEMENT') && (
+          <Security needs={[EXPLORE_EXUPDATE, INVESTIGATION_INUPDATE]}>
+            <WorkspaceCreation
+              paginationOptions={workspacePaginationOptions}
+              type={type}
+            />
+          </Security>
+        )}
+        taskScope={type === 'dashboard' ? 'DASHBOARD' : 'INVESTIGATION'}
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/private/components/workspaces/Workspaces.tsx:149]: What is the difference between type 'dashboard' and  'DASHBOARD'

**Review by marieflorescontact — COMMENTED** [review]
locally tested ok 👌
Do you plan to make some backend tests for `backgroundTask-common.js ` 🤓 ?

@lndrtrbn answer: for now I can't find any tests on background tasks, could be nice to add some but I don't know if we will have time to do it before release

File: opencti-platform/opencti-front/src/private/components/workspaces/Workspaces.tsx (hunk 1 above)
Comment (lndrtrbn) [reply opencti-platform/opencti-front/src/private/components/workspaces/Workspaces.tsx:149]: dashboard = a property of an entity `Workspace`, could be `dashboard` or `investigation`
DASHBOARD = the scope of the background task that is declared in our graphql

**Comment by Archidoit** [comment]
Filters not correctly displayed when launching a background task on public dashboards.

- Select all the public dashboard
- Click on delete in the toolbar
- the pop-up opens and the entity type public dashboard is not translated
image

**Comment by Archidoit** [comment]
Background task on custom dashboard also delete public dashboards : is it normal?

Select all the custom dashboards and delete them. Public dashboards are also deleted.

@lndrtrbn answer: Yes it is. Deleting a dashboard also delete its associated public dashboards.

**Comment by Archidoit** [comment]
Missing translations also for entity_type_Workspace

image

**Comment by Archidoit** [comment]
Export not working for Investigations

Select all the Investigations.

Export in stix. The export is working and the bundle is empty.

Export in Csv. An error occurs : 
image

**Comment by Archidoit** [comment]
Open the export panel of Custom dashboards. You also see the exports that have been done for Investigations.

**Comment by lndrtrbn** [comment]
@marieflorescontact @Archidoit remarks have been resolved

File: opencti-platform/opencti-graphql/tests/01-unit/domain/backgroundTask-common-test.ts [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+        operator: 'eq',
+        mode: 'or'
+      }
+    ],
+    filterGroups: []
+  });
+};
+
+describe('Background task validity check (checkActionValidity)', () => {
+  const userParticipate = buildStandardUser([], [], [{ name: 'KNOWLEDGE_KNPARTICIPATE' }]);
```
Comment (marieflorescontact) [inline opencti-platform/opencti-graphql/tests/01-unit/domain/backgroundTask-common-test.ts:46]: we have some users already build in testQuery.ts .weren't they enough?

Comment (marieflorescontact) [reply opencti-platform/opencti-graphql/tests/01-unit/domain/backgroundTask-common-test.ts:46]: thanks for the tests 🙂

Comment (lndrtrbn) [reply opencti-platform/opencti-graphql/tests/01-unit/domain/backgroundTask-common-test.ts:46]: The users created in testQuery.ts are in format to be ingested in the platform, not to be used as `context.user`. And I didn't want to make some queries to fetch them from elastic, was easier to create others

Comment (lndrtrbn) [reply opencti-platform/opencti-graphql/tests/01-unit/domain/backgroundTask-common-test.ts:46]: I tried at first to use those USER_EDITOR etc but it was not working

**Comment by Archidoit** [comment]
Public dashboard scope not translated in the Background tasks list 
image

**Comment by Archidoit** [comment]
In the background tasks list, try to delete a background task you previously do on Public Dashboard. An error occurs : 
image

**Review by Archidoit — APPROVED** [review]

**Pull request merged**